### DISCUSSION TITLE
Adding legacy versions of MKVToolNix

### DISCUSSION THREAD

**Discussion by whitebear60** [Homebrew/discussions#2767] [Writing Formulae/Casks] [answered]
Good evening.

The `mkvtoolnix` formula and cask supports only macOS 10.15+, however, the legacy versions of it had support of the older versions of macOS, up to OS X 10.9 Mavericks. Can that legacy versions be added to the `homebrew/cask-versions` tap, or similar software can't be allowed here due to Homebrew's policy regarding legacy casks and, therefore, should be in taps that hosted by someone else?

**Comment by carlocab** [comment, marked as answer]
We'll review a pull request that updates the existing cask to install older versions before Catalina. There are about a hundred examples of this in the cask repository. Just `grep` for `MacOS\.version <=`. For example:
```
❯ rg 'MacOS\.version <=' -l | head
Casks/araxis-merge.rb
Casks/avidemux.rb
Casks/sf-symbols.rb
Casks/far2l.rb
Casks/choosy.rb
Casks/r.rb
Casks/calibre.rb
Casks/openemu.rb
Casks/isyncr.rb
Casks/max.rb
```

**Comment by whitebear60** [comment]
Moritz hosts all versions of MKVToolNix on the dedicated file server, so it shouldn't be so hard. Had never contributed to the Homebrew repositories, will try writing that cask to support older macOS versions as well, hope it will work.

Anyway, thanks for your answer.

**Reply by carlocab** [reply]
It's not a bad time to get started. This might help: https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md

We're always happy to help you along if you get stuck. Feel free to open even an incomplete PR and we'll help you finish it off.

**Reply by whitebear60** [reply]
Probably something like [this](https://github.com/Homebrew/homebrew-cask/pull/117287)?

**Reply by carlocab** [reply]
💯

**Reply by whitebear60** [reply]
Oops... Seems like 10.10 version not working properly... [JSON API](https://formulae.brew.sh/api/cask/mkvtoolnix.json) says that yosemite version is 29, but it should be 24... It is because I have specified <=10.9, but should specify <=10.10 in first assertion, because Mavericks is no longer supported even on best-effort basis?

**Reply by carlocab** [reply]
Yes, that first assertion probably should be `<= "10.10"`. Nice catch! Thanks for checking up on your PR so carefully.

**Comment by neutric** [comment]
> The mkvtoolnix formula and cask supports only macOS 10.15+

@whitebear60 What makes you think that?

I didn't check how https://formulae.brew.sh/api/cask/mkvtoolnix.json is compiled, but the cask description currently states:

```
  if MacOS.version <= "10.10"
    version "24.0.0"
    sha256 "758da621d3a92358885333b767d64b024197a8147a339b1a0d14e938673452f9"
  elsif MacOS.version <= "10.11"
    version "29.0.0"
```
`if MacOS.version <= "10.10"` means for versions of macOS less than or equal `10.10`. So macOS up to and including `10.10/"Yosemite"` will get `24.0.0`. Everything up to and including `10.11/"El Capitan"`, except for all versions matched by the prior branch of the if statement, will get `29.0.0`. This effectively means that just `10.11` will get `29.0.0`. Did your PR https://github.com/Homebrew/homebrew-cask/pull/117323 mean to add Mavericks to the list of OS versions that are supported by `24.0.0`? If so, the PR can be closed because users on `10.9` are served `24.0.0` already.

**Reply by whitebear60** [reply]
First of all, this discussion have started before I've submitted that PR to add support of the older macOS versions to the MKVToolNix cask, at that moment the cask code was [this](https://github.com/Homebrew/homebrew-cask/blob/b77313bf26b3e7e01fb5440b21cf6f1359924ffe/Casks/mkvtoolnix.rb) and, obviously, don't support macOS older than 10.15.
Second, I think that you're misunderstanding something, because now the cask code [looks like this](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/mkvtoolnix.rb):
```
  if MacOS.version <= "10.9"
    version "24.0.0"
    sha256 "758da621d3a92358885333b767d64b024197a8147a339b1a0d14e938673452f9"
  elsif MacOS.version <= "10.11"
    version "29.0.0"
```

**Reply by neutric** [reply]
Apologies, you are correct. I started my msg by copy&paste on the phone and deleted the wrong line, which everything following is based on. Nothing to be seen here, move right along. ;)
